test/functional/http/post/3.votes.js

```diff
@@ -0,0 +1,464 @@
+					});
+				});
+			})
+			.then(function (res) {
+				var promisesCreditsMaxVotesPerAccount = [];
+				for (var i = 0; i < constants.activeDelegates; i++) {
+					var tempAccount = node.randomAccount();
+					delegatesMaxVotesPerAccount.push(tempAccount);
+					promisesCreditsMaxVotesPerAccount.push(creditAccountPromise(tempAccount.address, constants.fees.delegate));
+				};
+
+				return node.Promise.all(promisesCreditsMaxVotesPerAccount).then(function (results) {
+					results.forEach(function (res) {
+						node.expect(res).to.have.property('success').to.be.ok;
+						node.expect(res).to.have.property('transactionId').that.is.not.empty;
+					});
+				});
+			})
+			.then(function (res) {
+				return waitForBlocksPromise(7);
+			})
+			.then(function (res) {
+				return registerDelegatePromise(delegateAccount).then(function (result) {
+					node.expect(result).to.have.property('success').to.be.ok;
+					node.expect(result).to.have.property('transactionId').that.is.not.empty;
+				});
+			})
+			.then(function (res) {
+				var promisesDelegatesMaxVotesPerTransaction = [];
+				for (var i = 0; i < constants.maxVotesPerTransaction; i++) {
+					promisesDelegatesMaxVotesPerTransaction.push(registerDelegatePromise(delegatesMaxVotesPerTransaction[i]));
+				};
+
+				return node.Promise.all(promisesDelegatesMaxVotesPerTransaction).then(function (results) {
+					results.forEach(function (res) {
+						node.expect(res).to.have.property('success').to.be.ok;
+						node.expect(res).to.have.property('transactionId').that.is.not.empty;
+					});
+				});
+			})
+			.then(function (res) {
+				var promisesDelegatesMaxVotesPerAccount = [];
+				for (var i = 0; i < constants.activeDelegates; i++) {
+					promisesDelegatesMaxVotesPerAccount.push(registerDelegatePromise(delegatesMaxVotesPerAccount[i]));
+				};
+
+				return node.Promise.all(promisesDelegatesMaxVotesPerAccount).then(function (results) {
+					results.forEach(function (res) {
+						node.expect(res).to.have.property('success').to.be.ok;
+						node.expect(res).to.have.property('transactionId').that.is.not.empty;
+					});
+				});
+			})
+			.then(function (res) {
+				return waitForBlocksPromise(7);
+			});
+	});
+
+	describe('schema validations', function () {
+
+		shared.invalidAssets(delegateAccount, 'votes', badTransactions);
+	});
+
+	describe('transactions processing', function () {
+
+		it('voting with invalid vote format should fail', function () {
+			transaction = node.lisk.vote.createVote(delegateAccount.password, ['++' + node.eAccount.publicKey.slice(0, -1)]);
+
+			return sendTransactionPromise(transaction).then(function (res) {
+				node.expect(res).to.have.property('success').to.be.not.ok;
+				node.expect(res).to.have.property('message').to.equal('Invalid vote at index 0 - Invalid vote format');
+				badTransactions.push(transaction);
+			});
+		});
+
+		it('voting with invalid vote length should fail', function () {
```

Comment:
so the test is about voting with invalid length - don't include the special "--" operator here